File: src/pattern/observer/java/RandomNumberGenerator.java
package pattern.observer.java;

import java.util.Random;

public class RandomNumberGenerator extends NumberGenerator {
    Random random = new Random();
    private int number = 0;

    @Override
    public int getNumber() {
        return number;
    }

    @Override
    public void execute() {
        this.number = random.nextInt(50);
        this.notifyObservers();
    }
}

File: src/pattern/observer/kotlin/RandomNumberGenerator.kt
package pattern.observer.kotlin

import java.util.*

class RandomNumberGenerator: NumberGenerator(){
    val random: Random = Random()
    var _number: Int = 0

    override fun getNumber():Int{
        return this._number
    }

    override fun execute() {
        this._number = random.nextInt(50)
        this.notifyObservers()
    }

}